### DISCUSSION TITLE
Question: How to merge cropped PDF page with full pdf page in PyPDF?

### DISCUSSION THREAD

**Discussion by lananelson** [py-pdf/pypdf#717] [Q&A]
I am trying to merge two pdf pages using PyPDF2:

```
background_page = background_pdf.getPage(0)
overlay = some_pdf.getPage(0)

# crop overlaying page vertically
overlay.mediaBox.lowerRight = (
    overlay.mediaBox.getUpperRight_x(),
    overlay.mediaBox.getUpperRight_y() / 2
)
# merge
background_page.mergePage(overlay)
```

Unfortunately cropping doesn't happen. `mediaBox` of the overlaying page appears to be ignored and I see entire page in the output.

Does anyone have any idea of how I can achieve this?

I posted this on SO as well http://stackoverflow.com/questions/27512246/how-to-merge-cropped-pdf-page-with-full-pdf-page-in-pypdf

**Comment by mstamy2** [comment]
Try

```
background_page = background_pdf.getPage(0)
overlay = some_pdf.getPage(0)

# crop overlaying page vertically
overlay.mediaBox.lowerRight = (
    overlay.mediaBox.getUpperRight_x(),
    overlay.mediaBox.getUpperRight_y() / 2
)
# merge
overlay.mergePage(background_page) # Switched

output = PdfFileWriter()
output.addPage(overlay) # Add the Overlay Page
```

**Comment by lananelson** [comment]
If I merge pages this way then my overlay ends up being underneath the background. Is there a way to crop a page and save it in a new pdf without any data for the entire page? Right now if I crop a page using `mediaBox`  and write it to the output file I can easily get entire page again if I change the `mediaBox` back to its original values. I hope this makes sense. Thanks!

**Comment by martinmanzo** [comment]
Hi @lananelson, I'm having a similar problem but can't figure out how to solve it. Did you find a way around it?

I'm cropping a page `mediaBox` and saving the PDF. If I open the file using any standard PDF reader I see the content cropped the way I wanted, but if I print it I get the original page as if no cropping had occurred.

**Comment by lananelson** [comment]
@tincho89 I moved on from this a while ago and unfortunately don't exactly remember what the solution was (I don't have access to the repo anymore either). I think in your case you could try creating a blank page with the desired cropped dimensions and then merge your cropped page on top of it.

**Comment by martinmanzo** [comment]
@lananelson thanks for the response.

I was doing a proof of concept so I'm going to work on it again on some weeks, that approach looks good though. Will try it!

Thanks!

**Comment by forrestv** [comment]
See PR #240 for a fix.

**Reply by MartinThoma** [reply]
This was just merged and will be in the next release (1.27.6 - some time in April 2022)

**Comment by felle9900** [comment]
> Hi @lananelson, I'm having a similar problem but can't figure out how to solve it. Did you find a way around it?
> 
> I'm cropping a page `mediaBox` and saving the PDF. If I open the file using any standard PDF reader I see the content cropped the way I wanted, but if I print it I get the original page as if no cropping had occurred.

I have a similar problem - did you ever find a solution to your problem?

**Reply by martinmanzo** [reply]
Hi @felle9900 , I can't remember exactly, but I believe I ended up doing what @lananelson recommended.
